src/test/regress/expected/multi_transaction_recovery.out

```diff
@@ -331,6 +331,104 @@ SELECT count(*) FROM pg_dist_transaction;
      2
 (1 row)
 
+-- check that read-only participants skip prepare
+-- only two of the connections will perform a write (INSERT)
+SET citus.shard_count TO 4;
+CREATE TABLE test_2pcskip (a int);
+SELECT create_distributed_table('test_2pcskip', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO test_2pcskip SELECT i FROM generate_series(0, 5)i;
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions
+---------------------------------------------------------------------
+                             0
+(1 row)
+
+BEGIN;
+-- these inserts use two connections
+INSERT INTO test_2pcskip VALUES (6);
+INSERT INTO test_2pcskip VALUES (7);
+-- we know this will use more than two connections
+SELECT count(*) FROM test_2pcskip;
+ count
+---------------------------------------------------------------------
+     8
+(1 row)
+
+COMMIT;
+SELECT count(*) FROM pg_dist_transaction;
+ count
+---------------------------------------------------------------------
+     2
+(1 row)
+
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions
+---------------------------------------------------------------------
+                             0
+(1 row)
+
+-- check that read-only participants skip prepare
+-- only two of the connections will perform a write (INSERT)
+BEGIN;
+-- this insert uses two connections
+INSERT INTO test_2pcskip SELECT i FROM generate_series(8, 10)i;
```

Comment:
can we use the same values as we used before, 6 and 7 to make sure that these two values are on different shards/workers?